python/spglib/spglib.py

```diff
@@ -61,6 +63,16 @@
     from spglib import _spglib as spg  # type: ignore[attr-defined]
 
 
+# (lattice, positions, numbers) or (lattice, positions, numbers, magmoms)
+Lattice = Sequence[Sequence[float]]
+Positions = Sequence[Sequence[float]]
+Numbers = Sequence[int]
+Magmoms = Sequence[float] | Sequence[Sequence[float]]
+Cell: TypeAlias = (
+    tuple[Lattice, Positions, Numbers] | tuple[Lattice, Positions, Numbers, Magmoms]
```

Comment:
I think there's a better alternative here to also provide name hints

Reply:
Is it a comment or suggestion? Another way will be to narrow down `cell` into `typing.NamedTuple`.

Reply:
Probably I'm thinking of `NamedTuple`. I was hoping for something like TypedDict. The thing is that this one should work without having to use a NamedTuple constructor.

Although might be useful to provide a proper dataclass and we just forward the tuple/kwargs to that

Reply:
About changing the input type, yes it should be in a different PR. But for this PR, minimal type-hinting should still be added as I commented below.